legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/pureToSQLQuery/pureToSQLQuery.pure

```diff
@@ -4987,11 +4999,18 @@ function meta::relational::functions::pureToSqlQuery::processProject(ids:String[
          let nonOlap = $newColumnsWithPossiblePKs->filter(c|!$c->instanceOf(WindowColumn));
 
          let possiblePKs = $possiblePKsWithUpdatedData.first.values->cast(@Alias).relationalElement->cast(@TableAliasColumn).column->map(c| pair($c.name, ^PathInformation(type = $c.type->meta::relational::metamodel::datatype::dataTypeToCompatiblePureType(), relationalType = $c.type)));
+         
+         let importDataFlowCols = if($state.importDataFlow == true && $state.importDataFlowAddFks == true,
+                                     |$mainQuery.select.columns->cast(@Alias)->map(c| if($c.relationalElement->instanceOf(TableAliasColumn),
+                                                                                         |let aliases = $c.relationalElement->cast(@TableAliasColumn)->extractColumnAliasesInSelectQuery($merge, $context, $extensions);
+                                                                                          if($aliases->size() == 1, |^$c(relationalElement=$aliases->toOne()), |$c);,
+                                                                                         |$c)),
+                                     |[]);
```

Comment:
If I understand it correctly, in prcoessRelationalMappingSpecification, a TableAliasColumn is being generated with 'root' TableAlias and here its being changed to the correct TableAlias. This doesn't seem to be ideal.